peekaboo/sample.py

```diff
@@ -265,12 +288,23 @@ def sha256sum(self):
 
     @property
     def known(self):
-        _known = self.__db_con.known(self)
-        if _known:
+        if self.known_to_db:
             self.set_attr('known', True)
             return True
         return False
 
+    # These two hint at architectural breakage: Why do we sometimes want to know if
+    # a sample is known, its previous classification (result, reason) without
+    # updating its internal state (attribute)? Why is the sample interacting
+    # with the database in the first place?
```

Comment:
`known_to_db` is used by a rule to decide if an identical sample is analysied again or passed with the "old" result.

The idea behind this was to allow a decision by more granular criteria "configurable" in the ruleset (e.g. time since last analysis, age of vms, config changes, updates ...).

This will be more important in future when scoring is introduced to change indications into values with a threshold to make a decision rather than a single rule deciding.

For now this is totally fine

Reply:
I will leave it like this for now (with the comment) in the expectation that further refactoring will reveal where this logic and distinction fits in the architecture.